sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -6833,6 +6833,172 @@ public virtual Uri GenerateSasUri(BlobSasBuilder builder, out string stringToSig
+        /// and builder passed. The SAS is signed by the user delegation key
+        /// that is passed in.
+        ///
+        /// For more information, see
+        /// <see href="https://learn.microsoft.com/en-us/rest/api/storageservices/create-user-delegation-sas">
+        /// Creating an user delegation SAS</see>.
+        /// </summary>
+        /// <param name="builder">
+        /// Required. Used to generate a Shared Access Signature (SAS).
+        /// </param>
+        /// <param name="userDelegationKey">
+        /// Required. A <see cref="UserDelegationKey"/> returned from
+        /// <see cref="Azure.Storage.Blobs.BlobServiceClient.GetUserDelegationKeyAsync"/>.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Uri"/> containing the SAS Uri.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="Exception"/> will be thrown if a failure occurs.
+        /// </remarks>
+        [CallerShouldAudit("https://aka.ms/azsdk/callershouldaudit/storage-blobs")]
+        public virtual Uri GenerateUserDelegationSasUri(BlobSasBuilder builder, UserDelegationKey userDelegationKey) =>
+            GenerateUserDelegationSasUriInternal(builder, userDelegationKey, out _);
+
+        /// <summary>
+        /// The <see cref="GenerateUserDelegationSasUriInternal(BlobSasBuilder, UserDelegationKey, out string)"/>
+        /// returns a <see cref="Uri"/> representing a Blob Service
+        /// Shared Access Signature (SAS) Uri based on the Client properties
+        /// and builder passed. The SAS is signed by the user delegation key
+        /// that is passed in.
+        ///
+        /// For more information, see
+        /// <see href="https://learn.microsoft.com/en-us/rest/api/storageservices/create-user-delegation-sas">
+        /// Creating an user delegation SAS</see>.
+        /// </summary>
+        /// <param name="builder">
+        /// Required. Used to generate a Shared Access Signature (SAS).
+        /// </param>
+        /// <param name="userDelegationKey">
+        /// Required. A <see cref="UserDelegationKey"/> returned from
+        /// <see cref="Azure.Storage.Blobs.BlobServiceClient.GetUserDelegationKeyAsync"/>.
+        /// </param>
+        /// <param name="stringToSign">
+        /// For debugging purposes only.  This string will be overwritten with the string to sign that was used to generate the SAS Uri.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Uri"/> containing the SAS Uri.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="Exception"/> will be thrown if a failure occurs.
+        /// </remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [CallerShouldAudit("https://aka.ms/azsdk/callershouldaudit/storage-blobs")]
+        internal virtual Uri GenerateUserDelegationSasUriInternal(BlobSasBuilder builder, UserDelegationKey userDelegationKey, out string stringToSign)
```

Comment:
We don't need an internal version of this method.  Instead, we can put all the logic into the overload taking the `out string stringToSign`, and have the overload without the stringToSign call into it.

See:
- https://github.com/Azure/azure-sdk-for-net/blob/f2864e777b8f0067c3948909d8d18bba9c84bf64/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs#L3767
- https://github.com/Azure/azure-sdk-for-net/blob/f2864e777b8f0067c3948909d8d18bba9c84bf64/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs#L3797
- https://github.com/Azure/azure-sdk-for-net/blob/f2864e777b8f0067c3948909d8d18bba9c84bf64/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs#L3705
- https://github.com/Azure/azure-sdk-for-net/blob/f2864e777b8f0067c3948909d8d18bba9c84bf64/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs#L3741

Reply:
Hi @seanmcc-msft, I intentionally added the internal method after discussing with @amnguye. The reasoning is that the 2nd and 4th methods are hidden and never directly called by the customers anyways so I made them internal. The 1st and 3rd method are the ones exposed to the customer which eventually both flows to the 4th method that actually contains the logic.

Let me know if this makes sense and your thoughts

Reply:
There should be 4 methods total:

- `Uri GenerateUserDelegationSasUri(BlobContainerSasPermissions permissions, DateTimeOffset expiresOn, UserDelegationKey userDelegationKey)`
- `Uri GenerateUserDelegationSasUri(BlobContainerSasPermissions permissions, DateTimeOffset expiresOn, UserDelegationKey userDelegationKey, out string stringToSign)`
- `Uri GenerateUserDelegationSasUri(BlobSasBuilder builder, UserDelegationKey userDelegationKey)`
- `Uri GenerateUserDelegationSasUri(BlobSasBuilder builder, UserDelegationKey userDelegationKey, out string stringToSign)`

All 4 should be public, but the ones with `out stringToSign` should be EditorBrowsable Never.  All the actually logic for these methods should be contained in - `Uri GenerateUserDelegationSasUri(BlobSasBuilder builder, UserDelegationKey userDelegationKey, out string stringToSign)`